From 8c4b84f290e50e62fbf90a61df8c701140a42715 Mon Sep 17 00:00:00 2001
From: Thomas Baumgart <thb@net-bembel.de>
Date: Mon, 23 Jun 2025 10:15:20 +0200
Subject: [PATCH] Port triggering actions to Qt6

Qt6 requires a QAction to be enabled to be triggered programmatically.

BUG: 505977
FIXED-IN: 5.2.1
---
 kmymoney/kmymoney.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kmymoney/kmymoney.cpp b/kmymoney/kmymoney.cpp
index de03c0ddb7..582a391bfc 100644
--- a/kmymoney/kmymoney.cpp
+++ b/kmymoney/kmymoney.cpp
@@ -1090,7 +1090,10 @@ KMyMoneyApp::KMyMoneyApp(QWidget* parent)
 
     connect(d->m_myMoneyView, &KMyMoneyView::requestActionTrigger, this, [&](eMenu::Action action) {
         if (pActions.contains(action)) {
+            const bool enabled = pActions[action]->isEnabled();
+            pActions[action]->setEnabled(true);
             pActions[action]->trigger();
+            pActions[action]->setEnabled(enabled);
         }
     });
 
-- 
GitLab

